internal/core/model.rs

```diff
@@ -896,27 +896,30 @@ impl<C: RepeatedComponent + 'static> Repeater<C> {
         model: &ModelRc<C::Data>,
         count: usize,
     ) -> bool {
+        let mut indices_to_init = Vec::new();
         let mut inner = self.0.inner.borrow_mut();
         inner.components.resize_with(count, || (RepeatedComponentState::Dirty, None));
         let offset = inner.offset;
         let mut any_items_created = false;
         for (i, c) in inner.components.iter_mut().enumerate() {
             if c.0 == RepeatedComponentState::Dirty {
-                let created = if c.1.is_none() {
+                if c.1.is_none() {
                     any_items_created = true;
                     c.1 = Some(init());
```

Comment:
Can't this call to init() also recurse?

Reply:
Yes, true, there's probably a way of doing that.

Reply:
i think just by adding another layer:

```
// added if true there:
if true: layout := VerticalLayout {
        if true: TextInput {
            // Trigger a call into the layout
            init => { debug(self.y); }
        }
    }
```

But maybe self.y don't cross layers, but some other properties certainly will

Reply:
I wonder how to fix this. I mean, I made a local change to separate out the init() creation without borrowing the inner, but conceptually this would still recurse because the dirty() flag is true.

Perhaps like this?

1. Replace inner.components with an empty vec; set dirty to false.
2. Update any dirty components, call init, etc() - during that time any recursion will find an empty repeater.
3. Replace inner.components with the updated vec.